aimmo-game/tests/test_simulation/test_pickups.py

```diff
@@ -44,25 +47,85 @@ def test_effect_added(self):
 class TestHealthPickup(_BaseCases.BasePickupTestCase):
     pickup_class = pickups.HealthPickup
 
+    def test_invulnerability_is_applied(self):
+        self.apply_pickup()
+        self.assertIsInstance(self.pickup, pickups.HealthPickup)
+
     def test_health_increases(self):
         self.apply_pickup()
         self.assertEqual(self.avatar.health, 8)
 
     def test_serialise(self):
         self.assertEqual(self.pickup.serialise(), {'type': 'health', 'health_restored': 3})
 
+    def test_given_custom_health_pickup_increase_health_increases(self):
+        # Test 1 - increase of 5
+        self.assertEqual(self.avatar.health, 5)
+        self.set_custom_pickup(5)
+        self.apply_pickup()
+        self.assertEqual(self.avatar.health, 10)
+
+        # Test 2 - increase of 50
+        self.avatar.health = 10
+        self.set_custom_pickup(50)
+        self.apply_pickup()
+        self.assertEqual(self.avatar.health, 60)
+
+    def test_health_argument_raises_exception(self):
+        self.assertRaises(ValueError, self.set_custom_pickup, 150)
```

Comment:
This test case is testing three edge cases.
Ideally we want to have three tests each testing the specific edge case, otherwise it is just hiding the fact that there are edge cases.

test_health_argument_raise_exception_when_health_exceeded
etc ...

These tests make it clear what the edge cases are instead of grouping them based on their return value or thrown value. That isn't how tests should be written IMO.

Reply:
+1